static/js/components/AssignmentList.jsx

```diff
@@ -15,20 +15,21 @@ const AssignmentList = ({ assignments }) => {
   const { users } = useSelector((state) => state);
   const { observingRunList } = useSelector((state) => state.observingRuns);
   const { instrumentList } = useSelector((state) => state.instruments);
+  const [loadingUsers, setLoadingUsers] = useState([]);
 
   // fetch all the requester ids before rendering the component
   const requesterIDs = assignments.map((assignment) => assignment.requester_id);
-  const uniqueRequesterIDs = [...new Set(requesterIDs)];
-  uniqueRequesterIDs.sort((a, b) => a - b);
 
   // use useEffect to only send 1 fetchUser per User
   useEffect(() => {
-    uniqueRequesterIDs.forEach((id) => {
-      if (!users[id]) {
+    requesterIDs.forEach((id) => {
+      if (!users[id] && !loadingUsers.includes(id)) {
         dispatch(UserActions.fetchUser(id));
+        loadingUsers.push(id);
```

Comment:
I think modifying the state object directly is an anti-pattern; we should make a copy of `loadingUsers`, operate on that copy, and then pass it into `setLoadingUsers`